.github/workflows/debezium-workflow.yml

```diff
@@ -843,6 +843,63 @@ jobs:
           -Dmaven.wagon.http.pool=false
           -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
 
+  build_jdbc:
+    needs: [ check_style, file_changes ]
+    if: ${{ needs.file_changes.outputs.common-changed == 'true'}}
+    name: "JDBC"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Action (Core)
+        uses: actions/checkout@v3
+        with:
+          path: core
+
+      - name: Checkout Action (JDBC)
+        uses: actions/checkout@v3
+        with:
+          repository: debezium/debezium-connector-jdbc
+          path: jdbc
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: 17
+
+      # We explicitly use only the hash of the POM files from the core repository by default
+      # For this build, we do not care if there are or are not changes in the sibling repository since this
+      # job will only ever fire if there are changes in the common paths identified in the files_changed job.
+      - name: Cache Maven Repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: maven-debezium-test-build-${{ hashFiles('core/**/pom.xml') }}
+          restore-keys: |
+            maven-debezium-test-build-${{ hashFiles('core/**/pom.xml') }}
+
+      - name: Build Debezium (Core)
+        run: >
+          ./core/mvnw clean install -f core/pom.xml
+          -pl debezium-assembly-descriptors,debezium-bom,debezium-core,debezium-embedded,:debezium-ide-configs,:debezium-checkstyle,:debezium-revapi
+          -am
+          -DskipTests=true
+          -DskipITs=true
+          -Dcheckstyle.skip=true
+          -Dformat.skip=true
+          -Drevapi.skip
+          -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+          -Dmaven.wagon.http.pool=false
+          -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
+
+      - name: Build Debezium Connector JDBC
+        run: >
+          ./core/mvnw clean install -f jdbc/pom.xml -Passembly
+          -Dcheckstyle.skip=true
+          -Dformat.skip=true
+          -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+          -Dmaven.wagon.http.pool=false
+          -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
```

Comment:
So in the JDBC connector repository, there are two types of tests:

**Integration tests**
These tests leverage a factory to construct monk `SinkRecord` instances, and the JDBC connector consumes these mock records.  Therefore connector behavior changes in the main repository won't be reflected in these tests.

**End to End tests**
These tests rely on spinning up a `debezium/connect:nightly` image to run the actual connector inside Kafka Connect and to verify full end-to-end behavior from source to sink connector pipelines.  Since the nightly images require code to be merged to the `main` branch, PR changes won't be reflected in the image build and therefore we won't see any impact of the PR changes until some future time when the image is baked.

Overall, I'm inclined to suggest we shouldn't run any type of integration tests and this should specify the `-DskipITs` toggle, but I'd be interested in what others think.

Reply:
So it's just a check that changes on core doesn't breaks the compilation on jdbc sink

Reply:
Agreed, although even that list is quite tiny, mostly dependency versions from the BOM, `testcontainers` artifact, and some bits around configuration. :grin:

Reply:
ok, let's skip ITs right now. As mentioned above the approach using locally built image would bve better but I'd recommend to log it as a separate follow-up Jira.

Reply:
CI changed to skip IT and opened this follow-up https://issues.redhat.com/browse/DBZ-6383